src/main/java/com/semmle/jira/addon/config/upgrades/UpgradeTask4CreateConfigKeyField.java

```diff
@@ -0,0 +1,64 @@
+package com.semmle.jira.addon.config.upgrades;
+
+import com.atlassian.jira.component.ComponentAccessor;
+import com.atlassian.jira.issue.CustomFieldManager;
+import com.atlassian.jira.issue.context.GlobalIssueContext;
+import com.atlassian.jira.issue.fields.CustomField;
+import com.atlassian.plugin.spring.scanner.annotation.export.ExportAsService;
+import com.atlassian.sal.api.message.Message;
+import com.atlassian.sal.api.upgrade.PluginUpgradeTask;
+import com.semmle.jira.addon.util.Constants;
+import com.semmle.jira.addon.util.JiraUtils;
+import java.util.Collection;
+import java.util.Collections;
+import org.springframework.stereotype.Component;
+
+@ExportAsService(PluginUpgradeTask.class)
+@Component
+public class UpgradeTask4CreateConfigKeyField implements PluginUpgradeTask {
+
+  private static final int BUILD_NUMER = 4;
+
+  @Override
+  public Collection<Message> doUpgrade() throws Exception {
+
+    CustomFieldManager customFieldManager = ComponentAccessor.getCustomFieldManager();
+
+    CustomField customField =
+        customFieldManager.getCustomFieldObjectByName(Constants.CUSTOM_FIELD_NAME);
```

Comment:
Ideally we use the getCustomFieldObject**s**ByName method here and check if a custom field with the right name exists that is owned by our plugin.